docs/CODE_REVIEW.md

```diff
@@ -0,0 +1,145 @@
+# How to do code reviews for curl
+
+Anyone and everyone is encouraged and welcome to review code submissions in
+curl. This is a guide on what to check for and how to perform a successful
+code review.
+
+## All submissions should get reviewed
+
+All pull requests and patches submitted to the project should be reviewed by
+at least one experienced curl maintainer before that code is accepted and
+merged.
+
+## Let the tools and tests take the first rounds
+
+On initial pull requests, let the tools and tests do their job first and then
+start out by helping the submitter understand the test failures and tool
+alerts.
+
+## How to provide feedback to author
+
+Be nice. Ask questions. Provide examples or suggestions of improvements.
+Assume best intentions. Remember language barriers.
+
+All first-time contributors can become regulars. Let's help them go there.
+
+## Is this a change we want?
```

Comment:
Maybe we should add a section on ABI changes somewhere around here? If the PR changes an exported prototype then it can easily go on the no-fly zone

Reply:
Yes, very good point. That's certainly something to be aware of and look for in a review.

Reply:
I'm adding this new section:
-----
## API/ABI stability or changed behavior

Changing the API and the ABI may be fine in a change but it needs to be done
deliberately and carefully. If not, a reviewer must help the author to realize
the mistake.

curl and libcurl are similarly very strict on not modifying existing
behavior. API and ABI stability is not enough, the behavior should also remain
intact as far as possible.

Reply:
LGTM

Reply:
Maybe this could contain some more info about what API/ABI compatibility means?

For example: I plan to move some select/timeval related code to curlx functions in the future, but I am not yet sure how to do that, because I don't know how new functions affect API/ABI compatibility. On Windows it is probably quite easy, because (DLL) functions need to explicitly exported, but on Linux this is not the case, right?

Reply:
The ABI guideline is [here](https://github.com/curl/curl/blob/b71628b633de3287fa83360a2019f0b7023b3aa3/docs/libcurl/ABI) (and I just realized I should convert that to markdown) - I think it's better to keep that separate, but of course we also need to make sure it is accurate and helpful enough.